appcelerator/titanium_mobile
# [TIMOB-25541] Android: Use ContentProvider in EmailDialog

**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-25541
- https://jira.appcelerator.org/browse/TIMOB-23482

**Summary:**

- Fixes bug where EmailDialog fails to attach files on Android 7.0 and higher.
- Fixes bug where EmailDialog fails to attach files in Microsoft Outlook, regardless of Android OS version.
- Fixed by using Titanium's new content provider by switching file attachment URI schemes from "file:" to "content:".

**Test:**
1. Set up a project using the test code attached to [TIMOB-25541](https://jira.appcelerator.org/browse/TIMOB-25541).
2. Build and run on an Android 7.0 or newer device.
3. Tap on the "Send E-Mail" button.
4. Select the "GMail" app if asked.
5. Verify that 2 files are attached: "File1.txt" and "File2.txt"
6. Install and set up Microsoft "Outlook" on the device, if not already.
7. Repeat steps 2-5 above, but choose "Outlook" for step 4 instead.
